src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerdeletionservice/event/publisher/sns/SnsPublisherTest.kt

```diff
@@ -0,0 +1,114 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerdeletionservice.event.publisher.sns
+
+import com.amazonaws.services.sns.AmazonSNS
+import com.amazonaws.services.sns.model.PublishRequest
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.mockito.ArgumentCaptor
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import uk.gov.justice.hmpps.sqs.HmppsQueueService
+import uk.gov.justice.hmpps.sqs.HmppsTopic
+import java.time.Clock
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+
+class SnsPublisherTest {
+
+  val hmppsQueueService = mock<HmppsQueueService>()
+  val snsClient = mock<AmazonSNS>()
+  val clock = Clock.fixed(Instant.parse("2027-03-25T00:00:00.00Z"), ZoneId.systemDefault())
+
+  lateinit var snsPublisher: SnsPublisher
+
+  @Nested
+  @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+  inner class DomainEventTopicExists {
+    @BeforeEach
+    fun setup() {
+      val hmppsTopic = mock<HmppsTopic>()
+      whenever(hmppsQueueService.findByTopicId("domainevents")).thenReturn(hmppsTopic)
+      whenever(hmppsTopic.snsClient).thenReturn(snsClient)
+      snsPublisher = SnsPublisher(hmppsQueueService, ObjectMapper())
+    }
+
+    @Test
+    fun `send offender deletion event`() {
+      snsPublisher.sendOffenderDeletionEvent(
+        "someEventType", "someDescription", "AA1234A", setOf(1, 2, 3), 123, LocalDateTime.now(clock)
+      )
+
+      val captor: ArgumentCaptor<PublishRequest> = ArgumentCaptor.forClass(PublishRequest::class.java)
+      verify(snsClient).publish(captor.capture())
+
+      assertThat(captor.allValues[0].message.toString()).isEqualToIgnoringWhitespace(
+        """
+       {
+          "eventType":"someEventType",
+          "additionalInformation":{
+             "id":123
+          },
```

Comment:
I like seeing the JSON in the test, much better than having to dig out a file in the resources directory 👍 

You can also do this without an argument captor by using `check` - not sure it's any better but I quite like it, e.g. https://github.com/ministryofjustice/probation-offender-search-indexer/blob/871663c2d974a69770f3f66a6b1f3dd6a867be76/src/test/kotlin/uk/gov/justice/digital/hmpps/indexer/service/IndexQueueServiceTest.kt#L50